torch_xla/csrc/cross_replica_reduces.cpp

```diff
@@ -103,6 +105,31 @@ std::shared_ptr<torch::lazy::Value> CreateToken(
   return std::make_shared<torch::lazy::Value>(std::move(ir_value));
 }
 
+////////////////////////////////////////////////////////////////////////////////////
+// The traceable collectives integration follows here, listed in alphabetical
+// order. RFC: https://github.com/pytorch/pytorch/issues/93173
+////////////////////////////////////////////////////////////////////////////////////
+
+// tag is ignored as it's only used in PyTorch to provide backward compatibility
+// with the traditional process group API.
+at::Tensor all_reduce(const at::Tensor& self, c10::string_view reduceOp,
+                      c10::string_view /*tag*/, at::IntArrayRef /*ranks*/,
+                      int64_t /*group_size*/) {
+  TORCH_LAZY_FN_COUNTER("xla::");
+  auto self_tensor = bridge::GetXlaTensor(self);
+  // TODO(alanwaketan): Use ranks and group_size to generate groups. Currently
+  // we just suse {} as a workaround. Scale is always 1.0 here, and we always
+  // pin layout.
+  auto result = tensor_methods::all_reduce(self_tensor, GetReduceType(reduceOp),
+                                           /*scale*/ 1.0,
+                                           /*groups*/ {}, /*pin_layout*/ true);
```

Comment:
do you need some way to translate between our `tag rank stride` format and xla's groups format?

Reply:
Yea, will follow up with that later.